realm/src/main/java/io/realm/RealmObject.java

```diff
@@ -78,9 +85,14 @@
     protected Row row;
     protected Realm realm;
 
+    private final List<RealmChangeListener> listeners = new CopyOnWriteArrayList<RealmChangeListener>();
```

Comment:
the only pb here is once you remove `RealmObject` you need a marker interface, so the POJO could register/unregister listeners (this BTW could be an alternative of using an annotation @RealmClass, the interface allows to identify Realm POJO + ensure some common behaviour)